### PR TITLE
Create log data set monitoring docs

### LINKED ISSUE #3344: [Docs] Document the dataset health management page
Figma design [here ](https://www.figma.com/file/aaJNMsOklYhjwtu5TP7UDK/%5BLogs%2B%5D-Organize-and-Admin?type=design&node-id=2354-176010&mode=design&t=3Bf6MUI5n6oL8Puo-0#602346547)

Project ticket: https://github.com/elastic/observability-dev/issues/2887

Project Page: https://github.com/orgs/elastic/projects/1184/views/12

Because the system is set to accept all logs, users need to be able to monitor their datasets to find and fix any issues that occur in their log data. The information found on this page will serve as a foundation for the user to monitor the overall health of their datastreams and find any datastreams with issues and fix them. The datastream quality page will have two iterations, with the first being mostly informative and the second providing a flyout where the user can take action.

The focus shouldn’t be documenting the UI, though we can give an overview. It should be more on that data provided and why this data is useful.

Additional information and collaboration happening in [this document](https://docs.google.com/document/d/19yDDipWssiUhRoZDk9qWCLhuCcvmRIs6qEKSMHrPlMk/edit).

### PR DESCRIPTION
**Pull request by mdbirnstiehl**
Closes #3344 

Because the system is set to accept all logs, users need to be able to monitor their data sets to find and fix any issues that occur in their log data. The information found on this page will serve as a foundation for the user to monitor the overall health of their data sets and find any data sets with issues and fix them. The data set quality page will available through Stack Management.

### CODE REVIEW

**Comment by mdbirnstiehl** [comment]
@gbamparop  I've update the data set monitoring docs here to provide some general information and tips on investigating issues. Is there any additional guidance you think we should provide users at this point?

@awahab07 I added the privilege requirements [here ](https://github.com/elastic/observability-docs/pull/3960/files#diff-7bd542c3bede8f72347458825ebc56a3f0ffd9abd461d1b4ff4ee269f76610a3R18-R20)

File: docs/en/serverless/logging/monitor-datasets.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+id: serverlessObservabilityMonitorDatasets
+slug: /serverless/observability/monitor-datasets
+title: Monitor log data set quality
```
Comment (alaudazzi) [inline docs/en/serverless/logging/monitor-datasets.mdx:4]:
```suggestion
title: Monitor log dataset quality
```
This should probably be one word?
There are more occurrences in the rest of the doc.

**Review by alaudazzi — COMMENTED** [review]
Left a minor editing suggestion.

File: docs/en/serverless/logging/monitor-datasets.mdx (hunk 1 above)
Comment (mdbirnstiehl) [reply docs/en/serverless/logging/monitor-datasets.mdx:4]: Thanks for pointing that out. It's actually the opposite where I need to change the other occurrences to "data set". Lisa pointed out that the [branding guide](https://brand.elastic.co/302f66895/p/194a3b-writing-style-guide/t/889c93) has it as two words.

File: docs/en/serverless/logging/log-monitoring.mdx [hunk 2]
```diff
@@ -79,6 +79,13 @@ The following resources provide information on viewing and monitoring your logs:
 - <DocLink slug="/serverless/observability/discover-and-explore-logs">Discover and explore</DocLink>: Discover and explore all of the log events flowing in from your servers, virtual machines, and containers in a centralized view.
 - <DocLink slug="/serverless/observability/aiops-detect-anomalies">Detect log anomalies</DocLink>: Use ((ml)) to detect log anomalies automatically.
 
+## Monitor datasets
+
+The **Datasets** page provides an overview of your datasets and their quality.
```
Comment (yngrdyn) [inline docs/en/serverless/logging/log-monitoring.mdx:84]: Page is currently called `Data Set Quality`, should we align this?
image

File: docs/en/serverless/logging/monitor-datasets.mdx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+description: Monitor log data sets to find degraded documents.
+tags: [ 'serverless', 'observability', 'how-to' ]
+---
+
+<p><DocBadge template="technical preview" /></p>
+<p><DocBadge template="beta" /></p>
+
+The **Data Set Quality** page provides an overview of your log data sets.
+Use this information to get an idea of your overall log data set quality and find data sets that contain incorrectly parsed documents.
+Access the Datasets page from the main ((kib)) menu and go to **Stack Management** → **Data Set Quality*.
```
Comment (yngrdyn) [inline docs/en/serverless/logging/monitor-datasets.mdx:14]: should this be `data sets page`?

File: docs/en/serverless/logging/monitor-datasets.mdx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+{/* Update screenshot when changes are done <DocImage size="2" url="../images/logs-dataset-overview.png" alt="Screen capture of the data set overview" /> */}
+
+<DocCallOut title="Requirements">
+  Users with the `viewer` role can see the Datasets Quality summary. To see the Active Datasets and Estimated Data summaries, users need the `monitor` [index privilege](((ref))/security-privileges.html#privileges-list-indices) for the `logs-*-*` index.
```
Comment (yngrdyn) [inline docs/en/serverless/logging/monitor-datasets.mdx:19]: Should this be `Data sets Quality summary`?
image

File: docs/en/serverless/logging/monitor-datasets.mdx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+</DocCallOut>
+
+The quality of your data sets is based on the percentage of degraded documents in each data set.
+A degraded document in a data set contains the [`_ignored`](((ref))/mapping-ignored-field.html) property because one or more of its fields were ignored during indexing.
+Fields are ignored for a variety of reasons.
+For example, when the [`ignore_malformed`](((ref))/mapping-ignored-field.html.html) parameter is set to true, if a document field contains the wrong data type, the malformed field is ignored and the rest of the document is indexed.
+
+From the data set table, you'll find information for each data set such as its namespace, size, when the data set was last active, as well as the percentage of degraded docs.
```
Comment (yngrdyn) [inline docs/en/serverless/logging/monitor-datasets.mdx:27]: `Size` is not available in serverless yet, maybe we should remove the reference from here?

File: docs/en/serverless/logging/monitor-datasets.mdx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+### Investigate issues in individual data sets
+To investigate issues in data sets with poor or degraded quality:
+
+1. Find data sets with degraded documents using the **Degraded Docs** column of the data sets table.
+1. Click the percentage in the **Degraded Docs** column to open the data set in Logs Explorer.
+
+The **Documents** table in Logs Explorer is automatically filtered to show documents that were not parsed correctly.
+Under the *actions* column, you'll see the degraded document icon (<DocIcon type="indexClose" title="degraded document icon" />).
+
+Now that you know which documents contain ignored fields, examine them more closely to find the origin of the issue.
```
Comment (yngrdyn) [inline docs/en/serverless/logging/monitor-datasets.mdx:45]: This is being changed by @achyutjhunjhunwala at the moment.
He is creating a detailed section for the ignored fields. Please Aj, add more details here so we can have this part updated

File: docs/en/serverless/logging/monitor-datasets.mdx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+From the data set table, you'll find information for each data set such as its namespace, size, when the data set was last active, as well as the percentage of degraded docs.
+The percentage of degraded documents determines the data set's quality according to the following scale:
+
+* Good (<DocImage flatImage alt="Good icon" url="../images/green-dot-icon.png" />): 0% of the documents in the data set are degraded.
+* Degraded (<DocImage flatImage alt="Degraded icon" url="../images/yellow-dot-icon.png" />): Greater than 0% and up to 3% of the documents in the data set are degraded.
+* Poor (<DocImage flatImage alt="Poor icon" url="../images/red-dot-icon.png" />): Greater than 3% of the documents in the data set are degraded.
+
+Opening the details of a specific data set shows the degraded documents history, a summary for the data set, and other details that can help you determine if you need to investigate any issues.
+
+### Investigate issues in individual data sets
```
Comment (yngrdyn) [inline docs/en/serverless/logging/monitor-datasets.mdx:36]: Another way of investigate was added by @achyutjhunjhunwala. Now when you open the flyout of an individual dataset you have the opportunity to see an overview of ignored fields
image

File: docs/en/serverless/logging/view-and-monitor-logs.mdx [hunk 8]
```diff
@@ -77,4 +77,10 @@ The following actions help you filter and focus on specific fields in the log de
 * **Filter for value (<DocIcon type="plusInCircle" title="filter for value icon" />):** Show logs that contain the specific field value.
 * **Filter out value (<DocIcon type="minusInCircle" title="filter out value icon" />):** Show logs that do _not_ contain the specific field value.
 * **Filter for field present (<DocIcon type="filter" title="filter for present icon" />):** Show logs that contain the specific field.
-* **Toggle column in table (<DocIcon type="listAdd" title="toggle column in table icon" />):** Add or remove a column for the field to the main Logs Explorer table.
\ No newline at end of file
+* **Toggle column in table (<DocIcon type="listAdd" title="toggle column in table icon" />):** Add or remove a column for the field to the main Logs Explorer table.
+
+## View log dataset details
```
Comment (yngrdyn) [inline docs/en/serverless/logging/view-and-monitor-logs.mdx:82]: Should this be changed to `log data set details`?

File: docs/en/serverless/logging/view-and-monitor-logs.mdx [hunk 9]
```diff
@@ -77,4 +77,10 @@ The following actions help you filter and focus on specific fields in the log de
...
+
+From the main ((kib)) menu, go to **Stack Management** → **Logs datasets* to view more details about your datasets and monitor their overall quality.
```
Comment (yngrdyn) [inline docs/en/serverless/logging/view-and-monitor-logs.mdx:84]:
```suggestion
From the main ((kib)) menu, go to **Stack Management** → **Data Set Quality* to view more details about your datasets and monitor their overall quality.
```

File: docs/en/serverless/logging/view-and-monitor-logs.mdx [hunk 10]
```diff
@@ -77,4 +77,10 @@ The following actions help you filter and focus on specific fields in the log de
...
+
+Refer to <DocLink id="serverlessObservabilityMonitorDatasets">Monitor datasets</DocLink> for more on the monitoring your datasets.
```
Comment (yngrdyn) [inline docs/en/serverless/logging/view-and-monitor-logs.mdx:86]: `Monitor data sets`?

**Comment by yngrdyn** [comment]
Hey @mdbirnstiehl thanks for creating this PR, I was not able to check the preview, it has like an old version (without the changes) is that expected?

File: docs/en/serverless/logging/monitor-datasets.mdx (hunk 7 above)
Comment (gbamparop) [reply docs/en/serverless/logging/monitor-datasets.mdx:36]: This section is for individual data sets, I agree that we should mention the data set flyout with the new section, maybe another section can be added for issues in individual logs, mentioning the section in logs explorer.

**Comment by gbamparop** [comment]
@mdbirnstiehl will the stateful doc be added in another PR?

File: docs/en/serverless/logging/monitor-datasets.mdx (hunk 5 above)
Comment (mdbirnstiehl) [reply docs/en/serverless/logging/monitor-datasets.mdx:27]: Will this be available in 8.15 for stateful?

**Comment by mdbirnstiehl** [comment]
> @mdbirnstiehl will the stateful docs be added in another PR?

@gbamparop I was waiting until we finalized the serverless docs and then I was planning on basing the stateful docs off of them. I can either add them to this PR after the serverless docs are finalized or create a new PR. 

One question is whether or not there will be significant changes between the two that would require any large rewrites.

File: docs/en/serverless/logging/monitor-datasets.mdx (hunk 5 above)
Comment (yngrdyn) [reply docs/en/serverless/logging/monitor-datasets.mdx]: Yes, it will be available for stateful

**Comment by yngrdyn** [comment]
> One question is whether or not there will be significant changes between the two that would require any large rewrites.

Changes so far are not substancial, only around `size` column and summary, `Size` is not available on serverless at the moment, only stateful

File: docs/en/serverless/logging/monitor-datasets.mdx (hunk 6 above)
Comment (yngrdyn) [reply docs/en/serverless/logging/monitor-datasets.mdx:45]: This might not be ready before 8.15 FF (July 2nd)

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-bugs0pexh-elastic-dev.vercel.app

* Built with commit 6e551d323439d1a13e8544b3cb5cf34ceab456cc

> Issues? Visit #next-docs in Slack

**Comment by mdbirnstiehl** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by mdbirnstiehl** [comment]
> Hey @mdbirnstiehl thanks for creating this PR, I was not able to check the preview, it has like an old version (without the changes) is that expected?

@yngrdyn sorry I missed this initially. You can view the preview for serverless [here ](https://elastic-dot-co-docs-production-5q94d328k-elastic-dev.vercel.app/current/serverless/observability/monitor-datasets). I had forgotten to add the doc build label.

**Review by yngrdyn — DISMISSED** [review]
LGTM, the only thing is that screenshot is outdated.
It's showing `logs-*` instead of `logs-*-*`

**Review by alaudazzi — APPROVED** [review]
LGTM

**Pull request merged**